src/plugins/share/public/components/url_panel_content.tsx

```diff
@@ -269,8 +270,14 @@ export class UrlPanelContent extends Component<UrlPanelContentProps, State> {
     return this.updateUrlParams(formattedUrl);
   };
 
-  private getSnapshotUrl = () => {
-    const url = this.props.shareableUrl || window.location.href;
+  private getSnapshotUrl = (forSavedObject?: boolean) => {
+    let url = '';
+    if (forSavedObject && this.props.shareableUrlForSavedObject) {
+      url = this.props.shareableUrlForSavedObject;
```

Comment:
nit: how about setting `shareableUrlForSavedObject` to default value of '', and then this whole line can be simplified to: `url = this.props.shareableUrlForSavedObject`?

Reply:
Not sure I understand exactly what you propose.
Given a default value of `''` for `shareableUrlForSavedObject`, and this line changing to:

```ts
const url = this.props.shareableUrlForSavedObject ?? (this.props.shareableUrl) || window.location.href)
```

Then any snapshot url would depend on `shareableUrlForSavedObject`, which is not what I need. The `shareableUrlForSavedObject` should be used only when the `forSavedObject` flag is on, not every time a snapshot URL is asked.

Reply:
Ah fair, I thought `forSavedObject` was always set to true, other lines were collapsed 😬